### DISCUSSION TITLE
Linux上如何禁用IP？

### DISCUSSION THREAD

**Discussion by YummyCocoa** [YummyCocoa/nginx_shell#3] [Q&A] [answered]
Linux上如何禁用IP呢？

感觉亚马逊之类的防火墙太麻烦了，软件级别有什么简单的方案呢？

**Comment by Mintimate** [comment, marked as answer]
想要软件级别的控制IP禁用，那可以试试看`iptables`。

首先，`iptables`清除某条链，可以使用：
```bash
iptables -F [链名]
```
比如，清除默认的input：
```bash
iptables -F input
```
举个例子：
<img width="505" alt="image" src="https://github.com/YummyCocoa/nginx_shell/assets/43734212/ef995599-091b-4dd2-85b8-7ade6cad9e9d">

其次，如果禁用某个IP，建议先创建一条链：
```bash
iptables -t filter -N [自定义链]
```

之后，关联到自带的链，比如：
```bash
iptables -t filter -I INPUT -j [自定义链]
```

这个时候，我们想禁用IP，你可以：
```bash
iptables -t filter -D [自定义链] -s 119.233.233.233  -j DROP
```

具体的iptables使用，可以看看官方文档。或者问问ChatGPT。
